README.md

```diff
@@ -1,15 +1,67 @@
-# SlicerPipelines
-SlicerPipelines is an extension for 3DSlicer that offers the ability to create simple modules (aka pipelines) via a GUI interface with no coding knowledge needed.
+SlicerPipelines
+---------------
 
-In general, the SlicerPipelines extension is meant to help improve efficiency when dealing with simple workflows. Two main modules currently exist, a PipelineCreator module that allows for the creation of pipelines, and a PipelineCaseIterator module that allows for automated running of a pipeline over all files in a directory.
+SlicerPipelines is an extension for 3D Slicer for creating and executing simple modules (aka pipelines) via a GUI interface with no coding knowledge needed.
 
-![PipelineCreator module](Screenshots/1.png)
-![Select Module pop up](Screenshots/2.png)
+<p align="center">
+  <img src="Pipelines.png" alt="SlicerPipelines Logo"/>
+</p>
 
----
+## Overview
 
-SlicerPipelines is a Slicer extension made for SlicerSALT. To learn more about Slicer, SlicerSALT, and Slicer extensions, check out the following resources.
+The SlicerPipelines extension allows to create and execute simple linear workflows using the `PipelineCreator` and the `PipelineCaseIterator` modules.
+
+The design and implementation of this extension was initially motivated to simplify the creation of shape analysis workflow specific to the SlicerSALT community.
+
+## Features
+
+* **Simple**: Create and execute pipelines with a few clicks.
+* **Composable**: Pipelines are themselve modules available as "steps" to create new pipelines.
+* **Extensible**: Regular Slicer module modules, python functions, or executable can be registered as pipeline steps.
+
+## Modules
+
+| Name | Description |
+|------|-------------|
+| [PipelineCreator](PipelineCreator) | Allows for the creation of pipelines via a GUI interface. |
+| [PipelineCaseIterator](PipelineCaseIterator) | Allows for automated running of a pipeline over all files in a directory via a GUI interface |
+| [PipelineModules](PipelineModules) | Support the registration of "wrappers" providing the steps used in the creation of pipelines by the `PipelineCreator` module. |
+| [PipelineCLIBridge](PipelineCLIBridge) | Bridge the gap between the CLI C++ interface and the Python code of the `PipelineCreator`. |
+
+_:warning: This extension is under active development. Its content, API and behavior may change at any time. Additionally, you should consider the `PipelineModules` and `PipelineCLIBridge` modules as implementation details and not depends on there API. We mean it !_
```

Comment:
I don't necessarily agree with this comment. The PipelineModules CLIModuleWrapping is meant to be offered to other developers so they can use it to make pipeline of their CLI modules.

Example:
https://github.com/NIRALUser/MeshToLabelMap/blob/master/MeshToLabelMapPipeline/MeshToLabelMapPipeline.py

Reply:
That being said, `PipelineCLINow` and `PipelineCLI` in `PipelineModulesLib.CLIModuleWrapping` are probably the only functions that are meant to be the public.

Also, everything in `PipelineParameters.py` is meant to be public.

Reply:
Thanks for the review

### > re: disclaimer

After we integrate, we will likely have some more feedback and enter a phase of API stabilization. The idea is to manage community expectation outside of the module like `MeshToLabelMapPipeline` we are maintaining and updating ourselves.

I will then change the sentence from:

```rst
_:warning: This extension is under active development. Its content, API and behavior may change at any time. Additionally, you should consider the `PipelineModules` and `PipelineCLIBridge` modules as implementation details and not depends on there API. We mean it !_
```

to the less specific one:

```rst
_:warning: This extension is under active development. Its content, API and behavior may change at any time. We mean it !_
```

### > re: API and developer documentation

For now, I suggest to add the following markdown documents and reference them from the  top-level `README.md`:

```
Docs/user_guide/modules/pipelinecreator.md
Docs/user_guide/modules/pipelinecaseiterator.md
Docs/developer_guide/modules/pipelinemodules.md
Docs/developer_guide/modules/pipelineclibridge.md
```

For `pipelinecreator.md` and `pipelinecaseiterator.md`, consider looking at examples in https://github.com/Slicer/Slicer/blob/master/Docs/user_guide/modules/, for example [endoscopy.md](https://github.com/Slicer/Slicer/blob/master/Docs/user_guide/modules/endoscopy.md)

Indeed, I expect the UI will have minor changes.

For `pipelinemodules.md` and `pipelineclibridge.md`, you can look at examples at https://github.com/Slicer/Slicer/tree/master/Docs/developer_guide/modules. For these ones, I will keep things brief with a high level sentence and few links to relevant classes.

Example of content for `pipelinemodules.md`
```
# PipelineModules

This module is intended to provide the API for implementing module wrappers and registering them with the `` module.

## Type of wrappers

[...]
```

### Why this documentation structures ?

This is motivated by https://github.com/Slicer/Slicer/issues/6090

### > re: Wiki

Is there a Roadmap we could share with some the high level task that will be done ? Adding the roadmap to the wiki is sensible.

Reply:
### > re: API and developer documentation

For now, I suggest to add the following markdown documents and reference them from the  top-level `README.md`:

```
Docs/user_guide/modules/pipelinecreator.md
Docs/user_guide/modules/pipelinecaseiterator.md
Docs/developer_guide/modules/pipelinemodules.md
Docs/developer_guide/modules/pipelineclibridge.md
```

For `pipelinecreator.md` and `pipelinecaseiterator.md`, consider looking at examples in https://github.com/Slicer/Slicer/blob/master/Docs/user_guide/modules/, for example [endoscopy.md](https://github.com/Slicer/Slicer/blob/master/Docs/user_guide/modules/endoscopy.md)

Indeed, I expect the UI will have minor changes.

For `pipelinemodules.md` and `pipelineclibridge.md`, you can look at examples at https://github.com/Slicer/Slicer/tree/master/Docs/developer_guide/modules. For these ones, I will keep things brief with a high level sentence and few links to relevant classes.

Example of content for `pipelinemodules.md`
```
# PipelineModules

This module is intended to provide the API for implementing module wrappers and registering them with the `` module.

## Type of wrappers

[...]
```

Reply:
### Why this documentation structures ?

This is motivated by https://github.com/Slicer/Slicer/issues/6090

Reply:
### > re: Wiki

Is there a Roadmap we could share with some the high level task that will be done ? Adding the roadmap to the wiki is sensible.

Reply:
While we should not wait for all of these items to be addressed before integrating in the extensions index, I suggest you create issues as you see fit.

To help with that, each comment is on its own, and you should be able to leverage the `Reference in new issue` feature:

![image](https://user-images.githubusercontent.com/219043/153304313-3dd0341f-fdef-4228-8f1b-3ddd2281d1ec.png)

Reply:
Issues have been added for the roadmap, the user guides, and the developer guides.